File: patterns/aggregator/java-camel/AggregatorRoute.java
import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.stream.Collectors;

public class AggregatorRoute {
    /**
     * Aggregate quotes while counting duplicates.
     * <p>
     * Example output: {@code hello(2),world(1)}
     */
    public static String aggregateQuotes(String[] quotes) {
        Map<String, Long> counts = Arrays.stream(quotes)
                .collect(Collectors.groupingBy(q -> q, LinkedHashMap::new, Collectors.counting()));
        return counts.entrySet().stream()
                .map(e -> e.getKey() + "(" + e.getValue() + ")")
                .collect(Collectors.joining(","));
    }

    public static void main(String[] args) {
        String[] quotes = {"hello", "world", "hello"};
        System.out.println(aggregateQuotes(quotes));
    }
}

File: patterns/aggregator/kotlin-spring-boot/AggregatorRoute.kt
object AggregatorRoute {
    /**
     * Aggregates quotes and counts duplicates.
     * Example output: {@code hello(2),world(1)}
     */
    fun aggregateQuotes(quotes: List<String>): String =
        quotes.groupingBy { it }
            .eachCount()
            .map { "${it.key}(${it.value})" }
            .joinToString(",")

    @JvmStatic
    fun main(args: Array<String>) {
        val quotes = listOf("hello", "world", "hello")
        println(aggregateQuotes(quotes))
    }
}
